agent/recorder.go

```diff
@@ -189,9 +237,9 @@ func (r *SpanRecorder) callIngest(payload io.Reader) (statusCode int, err error)
 
 	resp, err := r.client.Do(req)
 	if err != nil {
-		r.koSend++
 		return 0, err
 	}
+	io.Copy(ioutil.Discard, resp.Body) // for KeepAlive we need to be sure to read all body before close
```

Comment:
We should `defer` this call, as we are actually reading the body in case of an error (line 246)

Reply:
yes, definitely

Reply:
When moving the retry logic to this func I changed all this logic.